src/hotspot/cpu/aarch64/codeBuffer_aarch64.cpp

```diff
@@ -26,6 +26,61 @@
 #include "asm/codeBuffer.inline.hpp"
 #include "asm/macroAssembler.hpp"
 
+void CodeBuffer::share_trampoline_for(address dest, int caller_offset) {
+  if (_shared_trampoline_requests == nullptr) {
+    _shared_trampoline_requests = new SharedTrampolineRequests();
+  }
+  SharedTrampolineRequest request(dest, caller_offset);
+  _shared_trampoline_requests->push(request);
+  _finalize_stubs = true;
+}
+
+static bool emit_shared_stubs_to_runtime_call(CodeBuffer* cb, CodeBuffer::SharedTrampolineRequests* requests) {
+  if (requests == NULL) {
+    return true;
+  }
+
+  const int length = requests->length();
+  const int table_length = length * 2 + 1;
+  const int UNUSED = length;
+  intArray last(table_length, table_length, UNUSED);  // maps an dest address to the last request of the address
+  intArray prev(length, length, UNUSED);              // points to the previous request with the same dest address
+
+  for (int i = 0; i < length; i++) {
+    const address dest = requests->at(i).dest();
+    int j = intptr_t(dest) % table_length;
+    for (; last.at(j) != UNUSED && requests->at(last.at(j)).dest() != dest; j = (j + 1) % table_length)
+      ;
+    prev.at(i) = last.at(j);
+    last.at(j) = i;
+  }
```

Comment:
Why not to implement `SharedTrampolineRequests` as a simple hash table?
It can be based on: `GrowableArray<Pair<address, LinkedListImpl<int>>>`

Is it writing simple  hash table code better than using sorting? If a number of trampolines is small there should not be much performance difference. IMHO, less code is better. How many trampolines per method are created?